dot/digest/digest.go

```diff
@@ -139,42 +158,19 @@ func (h *Handler) handleConsensusDigest(d *types.ConsensusDigest, header *types.
 		if err != nil {
 			return err
 		}
-		err = h.handleGrandpaConsensusDigest(data, header)
-		if err != nil {
-			return err
-		}
-		return nil
+
+		return h.grandpaState.HandleGRANDPADigest(header, data)
 	case types.BabeEngineID:
 		data := types.NewBabeConsensusDigest()
 		err := scale.Unmarshal(d.Data, &data)
 		if err != nil {
 			return err
 		}
-		err = h.handleBabeConsensusDigest(data, header)
-		if err != nil {
-			return err
-		}
-		return nil
-	}
-
-	return errors.New("unknown consensus engine ID")
-}
 
-func (h *Handler) handleGrandpaConsensusDigest(digest scale.VaryingDataType, header *types.Header) error {
-	switch val := digest.Value().(type) {
-	case types.GrandpaScheduledChange:
-		return h.handleScheduledChange(val, header)
-	case types.GrandpaForcedChange:
-		return h.handleForcedChange(val, header)
-	case types.GrandpaOnDisabled:
-		return nil // do nothing, as this is not implemented in substrate
-	case types.GrandpaPause:
-		return h.handlePause(val)
-	case types.GrandpaResume:
-		return h.handleResume(val)
```

Comment:
because we don't support pause/resume digest messages yet, but there is an issue to implement them https://github.com/ChainSafe/gossamer/issues/2556 (I started, but this is not a right priority since substrate doesn't support them as well)